src/CoreWCF.Http/src/CoreWCF/Channels/RequestDelegateHandler.cs

```diff
@@ -97,6 +98,21 @@ private WebSocketOptions CreateWebSocketOptions(HttpTransportBindingElement tbe)
 
         internal async Task HandleRequest(HttpContext context)
         {
+            if (SecurityUtils.ShouldUseAuthentication(_serviceDispatcher.Binding))
+            {
```

Comment:
As mentioned elsewhere, ShouldUseAuthentication should live in this class and be driven by _httpSettings.AuthenticationScheme.